src/vos/tests/vts_checksum.c

```diff
@@ -405,6 +405,92 @@ update_fetch_csum_for_array_10(void **state)
 	});
 }
 
+static void
+mark_sv_corrupted(void **state)
+{
+	struct extent_key	 k = {0};
+	daos_iod_t		 iod = {0};
+	d_sg_list_t		 sgl = {0};
+	int			 rc = 0;
+
+	/** setup */
+	extent_key_from_test_args(&k, *state);
+	dts_sgl_init_with_strings(&sgl, 1, "Going to be corrupted!!");
+
+	iod.iod_name = k.akey;
+	iod.iod_size = daos_sgl_buf_size(&sgl);
+	iod.iod_nr = 1;
+	iod.iod_type = DAOS_IOD_SINGLE;
+
+	assert_success(vos_obj_update(k.container_hdl, k.object_id, 1, 0, 0,
+				      &k.dkey, 1, &iod, NULL, &sgl));
+
+	/** reset sgl and fetch just to sanity check that it works before
+	 * marking as corrupted
+	 */
+	memset(sgl.sg_iovs[0].iov_buf, 0, sgl.sg_iovs[0].iov_buf_len);
+	assert_success(vos_obj_fetch(k.container_hdl, k.object_id, 1, 0,
+				     &k.dkey, 1, &iod, &sgl));
+
+	/** Mark as corrupted using the corrupt flag */
+	assert_success(vos_obj_update(k.container_hdl, k.object_id, 1, 0,
+				      VOS_OF_CORRUPT,
+				      &k.dkey, 1, &iod, NULL, &sgl));
```

Comment:
I'm wondering how this "mark corrupted" API will be used, it inserts a special extent in evtree to mark a logical extent as corrupted? How will this 'corrupted' extent being processed by fetch or aggregation?

Since the checksum is per physical extent, isn't 'corrupted' per physical extent?

Reply:
It will update the bio_addr that points to the physical extent to have the corrupted flag. It takes advantage of the overwrite capability of evt_insert. vos_obj_update with the corrupted flag should not have an sgl with it. I've updated the tests a little and added documentation hopefully explaining how this works.